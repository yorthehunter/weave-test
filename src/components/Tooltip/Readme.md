Tooltip
```js
<Tooltip content="Hi, I'm a tooltip!">Hover here.</Tooltip>
```

Tooltip with hover indication
```js
<Tooltip hasIndicator content="Hi, I'm a tooltip!">Hover here.</Tooltip>
```

Tooltip below
```js
<Tooltip position="below" content="Hi, I'm a tooltip!">Hover here.</Tooltip>
```
