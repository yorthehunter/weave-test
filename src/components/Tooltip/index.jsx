import React from 'react';
import { string, bool, node } from 'prop-types';
import classNames from 'classnames';
import { positionModifier } from '../utilities/util';

const componentPrefix = 'weave-has-tooltip';


/**
 * Tooltips show very _short_ messages when content is hovered. This content
 * is usually a hint at a function or brief description of something.
 */

const Tooltip = (props) => {
  const {
    content,
    children,
    position,
  } = props;

  const componentClasses = classNames([
    componentPrefix,
    positionModifier(position, componentPrefix),
    props.hasIndicator && `${componentPrefix}-indication`,
  ]);

  return (
    <span
      className={componentClasses}
      data-tooltip={content}
    >
      { children }
    </span>
  );
};

const defaultProps = {
  content: null,
  hasIndicator: false,
  position: null,
  children: null,
};

const propTypes = {
  content: string,
  position: string,
  hasIndicator: bool,
  children: node,
};

Tooltip.defaultProps = defaultProps;
Tooltip.propTypes = propTypes;
export default Tooltip;
