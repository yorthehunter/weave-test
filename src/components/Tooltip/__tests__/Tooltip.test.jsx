import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Tooltip from '../';

describe('Tooltip', () => {
  const tooltipClass = '.weave-has-tooltip';
  const component = <Tooltip content="tooltip goes here!">This text holds a tooltip!</Tooltip>;
  const wrapper = mount(component);

  it('renders correctly with default values', () => {
    const rendered = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('correctly handles positioning', () => {
    wrapper.setProps({ position: 'below' });
    expect(wrapper.find(tooltipClass).hasClass('weave-has-tooltip--below')).toEqual(true);
  });

  it('correctly displays click indication', () => {
    wrapper.setProps({ hasIndicator: true });
    expect(wrapper.find(tooltipClass).hasClass('weave-has-tooltip-indication')).toEqual(true);
  });
});
