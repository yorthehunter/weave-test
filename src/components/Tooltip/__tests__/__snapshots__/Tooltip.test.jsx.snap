// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders correctly with default values 1`] = `
<span
  className="weave-has-tooltip"
  data-tooltip="tooltip goes here!"
>
  This text holds a tooltip!
</span>
`;
